brutils/currency.py

```diff
@@ -1,92 +1,70 @@
 from decimal import ROUND_DOWN, Decimal, InvalidOperation
-from typing import Union
 
 from num2words import num2words
 
 
-def format_currency(value):  # type: (float) -> str | None
+def format_currency(value: float | int | str | Decimal) -> str | None:
     """
-    Formats a given float as Brazilian currency (R$).
-
-    This function takes a float value and returns a formatted string representing
-    the value as Brazilian currency, using the correct thousand and decimal separators.
+    Format a numeric value as Brazilian currency (R$).
 
     Args:
-        value (float or Decimal): The numeric value to be formatted.
+        value: The numeric value to format. Accepts float, int, str, or Decimal.
 
     Returns:
-        str or None: A string formatted as Brazilian currency, or None if the input is invalid.
+        Formatted currency string (e.g., "R$ 1.234,56") or None if invalid.
 
-    Example:
+    Examples:
```

Comment:
Eu prefiro `Examples` também, pensando agora. Mas tá em todo lugar com `Example`. Então para manter o padrão eu deixaria sem o S e depois trocaria tudo de uma vez. Faz sentido? Ou nada a ver e bora dale no melhor mesmo?

```suggestion
    Example:
```

Reply:
vamos seguir o padrão e depois renomeamos geral.